### PR TITLE
ComboBox: support different DataTemplate for selected item.

### PR DESCRIPTION
**Pull request by rabbitism**
## What does the pull request do?

Add a new Property SelectedItemTemplate, if set, it has a higher priority than ItemTemplate.

## What is the current behavior?

Displayed selected item is always the same as ComboBoxItem

## What is the updated/expected behavior with this PR?

Make it possible to use different DataTemplates for selected item and other items.

image

## How was the solution implemented (if it's not obvious)?

Use multibinding and select the first valid IDataTemplate.

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

File: src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using System.Globalization;
+using Avalonia.Controls.Templates;
+using Avalonia.Data.Converters;
+
+namespace Avalonia.Controls.Converters;
+
+/// <summary>
+/// Select a valid IDataTemplate in order.
+/// </summary>
+public class MultiDataTemplatesConverter: IMultiValueConverter
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs:12]: I am not sure about multibinding approach. It makes whole feature highly dependent on a template, and well as adds a very specific converter to the public API.

Comment (rabbitism) [reply src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs:12]: Yes, I think  this is over designed. Maybe we can just remove ContentTemplate local value from ComboBox template, and set it in root style?

**Comment by MrJul** [comment]
Can we please call this `SelectionBoxItemTemplate` (same name as WPF, even if it's read-only there)?

My first thought while reading this PR was that the `SelectedItemTemplate` property also applies to the selected item _inside_ the dropdown, which it doesn't.

File: src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs (hunk 1 above)
Comment (MrJul) [reply src/Avalonia.Controls/Converters/MultiDataTemplatesConverter.cs:12]: You could also use the coercion callback to coerce this new property to `ItemTemplate` when it's null. This should simplify the XAML side with a classic single `{TemplateBinding}`.

**Comment by rabbitism** [comment]
Updated according to reviews.

**Comment by Gillibald** [comment]
Other controls call this property ContentTemplate

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**